src/core/transport/chttp2/hpack_encoder.c

```diff
@@ -461,6 +489,41 @@ void grpc_chttp2_hpack_compressor_destroy(grpc_chttp2_hpack_compressor *c) {
     if (c->entries_elems[i]) GRPC_MDELEM_UNREF(c->entries_elems[i]);
   }
   GRPC_MDSTR_UNREF(c->timeout_key_str);
+  gpr_free(c->table_elem_size);
+}
+
+void grpc_chttp2_hpack_compressor_set_max_usable_size(
+    grpc_chttp2_hpack_compressor *c, gpr_uint32 max_table_size) {
+  c->max_usable_size = max_table_size;
+  grpc_chttp2_hpack_compressor_set_max_table_size(
+      c, GPR_MIN(c->max_table_size, max_table_size));
+}
+
+static void rebuild_elems(grpc_chttp2_hpack_compressor *c, gpr_uint32 new_cap) {
+
```

Comment:
It's subtle: not having that function present will lead to memory
corruption eventually, but it's a very long eventually.

On Wed, Nov 18, 2015 at 3:52 PM David G. Quintas <notifications@github.com>
wrote:

> In src/core/transport/chttp2/hpack_encoder.c
> <https://github.com/grpc/grpc/pull/4097#discussion_r45279935>:
>
> > @@ -461,6 +489,41 @@ void grpc_chttp2_hpack_compressor_destroy(grpc_chttp2_hpack_compressor *c) {
> >      if (c->entries_elems[i]) GRPC_MDELEM_UNREF(c->entries_elems[i]);
> >    }
> >    GRPC_MDSTR_UNREF(c->timeout_key_str);
> > +  gpr_free(c->table_elem_size);
> > +}
> > +
> > +void grpc_chttp2_hpack_compressor_set_max_usable_size(
> > +    grpc_chttp2_hpack_compressor *c, gpr_uint32 max_table_size) {
> > +  c->max_usable_size = max_table_size;
> > +  grpc_chttp2_hpack_compressor_set_max_table_size(
> > +      c, GPR_MIN(c->max_table_size, max_table_size));
> > +}
> > +
> > +static void rebuild_elems(grpc_chttp2_hpack_compressor *c, gpr_uint32 new_cap) {
> > +
>
> so much for my review... well, I caught it, albeit for the wrong reasons.
> Maybe tests should be more exhaustive too :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/4097/files#r45279935>.
>